File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/sensors/Vision.java
package com.example.collectoragentmind.codelets.sensors;

import com.example.collectoragentmind.CollectorAgentMind;

import java.util.List;

import br.unicamp.cst.core.entities.Codelet;
import br.unicamp.cst.core.entities.MemoryObject;

public class Vision extends Codelet {

    /*
        Requests the apples in sight every proc. The body
        should respond to this request by updating the
        visionMO through the plugin's interface
    */

    private CollectorAgentMind plugin;

    public Vision() {
        plugin = CollectorAgentMind.getInstance();
    }

    @Override
    public void accessMemoryObjects() {
    }

    @Override
    public void proc() {
        plugin.emitRequest("request_apples_in_sight");
    }

    @Override
    public void calculateActivation() {

    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/sensors/Vision.kt
package com.example.game.cst.codelets.sensors

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.CollectorMindCommunicator

class Vision : Codelet() {
    /*
        Requests the apples in sight every proc. The body
        should respond to this request by updating the
        visionMO through the plugin's interface
    */

    private var visionMO: MemoryObject? = null

    override fun accessMemoryObjects() {
        visionMO = getInput("VISION") as MemoryObject?
    }

    override fun proc() {
        visionMO?.setI(CollectorMindCommunicator.instance.getApplesInVision())
    }

    override fun calculateActivation() {}
}